File: Algorithm/1_twoSum/hashmap/main.py

from pip import List


def twoSum(seft, nums: List[int], target: int) -> List[int]:
    temp = {}
    for positionOfNum in range(len(nums)):
        valueOfNum = nums[positionOfNum]
        if valueOfNum in temp:
            return [temp[valueOfNum], positionOfNum]
        difFromTarget = target - nums[positionOfNum]
        temp[difFromTarget] = positionOfNum
    return [-1, -1]


nums = [2, 3, 5, 7]
target = 9
ans = twoSum(0, nums, target)
print(ans[0], ans[1])

File: Algorithm/1_twoSum/hashmap/main.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> twoSum(vector<int> &nums, int target)
{
    map<int, int> m;
    for (int i = 0; i < nums.size(); i++)
    {
        if (m.find(nums[i]) != m.end())
        {
            return {m[nums[i]], i};
        }
        m[target - nums[i]] = i;
    }
    return {-1, -1};
}
int main()
{
    vector<int> a = {2, 7, 11, 15};
    vector<int> b = twoSum(a, 9);
    cout << b[0] << " " << b[1] << endl;
    return 0;
}